File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/service/arquivo/IArquivoService.java
package br.com.blu.bs2.sintetizador.contas.csv.service.arquivo;

import br.com.blu.bs2.sintetizador.contas.csv.model.arquivos.Arquivo;
import br.com.blu.bs2.sintetizador.contas.csv.utils.exceptions.ServiceException;

import java.io.File;
import java.util.List;

public interface IArquivoService {

    List<Arquivo> recuperarArquivosCsvFromPath(String path) throws ServiceException;

    List<Arquivo> recuperarArquivosCsvFromFilePath(File path) throws ServiceException;

    List<File> findAllFiles(File filePath) throws ServiceException;

    List<File> findAllFilesPor(String filePath) throws ServiceException;
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/service/file/IArquivoService.kt
package br.com.blu.bs2.sintetizador.contas.csv.service.file

import br.com.blu.bs2.sintetizador.contas.csv.model.arquivos.Arquivo
import br.com.blu.bs2.sintetizador.contas.csv.utils.exceptions.ServiceException
import java.io.File

interface IArquivoService {
    @Throws(ServiceException::class)
    fun recuperarArquivosCsvFromPath(path: String?): List<Arquivo?>?

    @Throws(ServiceException::class)
    fun recuperarArquivosCsvFromFilePath(path: File?): List<Arquivo?>?

    @Throws(ServiceException::class)
    fun findAllFiles(filePath: File?): List<File?>?

    @Throws(ServiceException::class)
    fun findAllFilesPor(filePath: String?): List<File?>?
}
